string instruction
bool fast_sam
---
int32 cX
int32 cY
